### ISSUE TITLE
301 and 302 Vanity URL redirects do not pass querystrings

### ISSUE THREAD

**Issue by craigWagner99** [dotCMS/core#19564]
As reported in https://dotcms.zendesk.com/agent/tickets/102845

github issue: https://github.com/dotCMS/core/issues/18325

That issue is closed but the issue persists.

If you wish to pass a querystring parameter in dotcms 5.3x+ via a Vanity URL the only way to do it is via a 200 Forward.

301 and 302 Redirects do not pass the querystring.

**To Reproduce**
1. Create a Vanity URL
e.g.
**Uri**: /store/test(.*)
**Action**: 301 - Permanent Redirect
**Forward To**: /blog/test$1

2. Navigate to the 'from' URL: 
demo.dotcms.com/store/test?foo=bar

**Expected behavior**
Page should redirect to 
**demo.dotcms.com/blog/test?foo=bar**

**Actual Behavior**
If you use 301 or 302 Redirect it does not.

If you use 200 - Forward it DOES work, BUT the URL remains:
**demo.dotcms.com/store/test?foo=bar**

THE 'WORKING' CONFIGURATION
![image](https://user-images.githubusercontent.com/49727631/98419399-9be3d300-204a-11eb-95db-2b12691873aa.png)

**Comment by wezell** [comment]
This is not a bug, rather that dotCMS was never designed to match query parameters and pass them to redirects.  It would be a new feature.

**Comment by john-thomas-dotcms** [comment]
To be clear, the issue is not that the vanity doesn't match the URL param - it's that even when the vanity doesn't use any regex at all, the URL param is not passed through to the page the vanity redirects so.  And this only happens with 301 and 302 redirects - with 200 redirects it does pass the URL param.

The original github issue to handle this (https://github.com/dotCMS/core/issues/18325) did not specify that there would be a difference between 200 and 301/302 requests, so that made it appear that this is a bug, not a feature request.

So @wezell is this really a feature request, or is it something we missed when we implemented #18325?

**Comment by rahulbpatel** [comment]
Following up on this.  What would it take to:

# Make 301 and 302 redirects behave similar to 200 so that query strings are included
# The ability to specify a query string variable in the Forward To to help with tracking redirects. i.e `/new-url?_redirectedFrom=%2FoldUrl`

**Comment by fabrizzio-dotCMS** [comment]
NOTE FOR QA:
scenarios that must be tested: 

Create VanityURL where the `Forward To` URI has fixed params.
like `/destinations/colorado?foo=bar&vanityParam=2`

And don't forget the incoming uri has to have the pattern to match new parts
like:/redirect1(.*)

The resulting URL has to have both params combined the fixed params and the ones passed on the incoming URL.

You need to test scenarios where the target "forward to" the page is both an internal and also an external resource.

Take into account that forwarding to a page does not necessarily change the URL. (Redirects do change the url)
But that doesn't mean the params aren't being passed. if you want to verify that params are truly getting passed you will need to use a Velocity page as the destination to verify so. 

something like this will do.

```
#if($UtilMethods.isSet($request.getParameter("myTestParam")))
    #set($myTestParam   = $!request.getParameter("myTestParam"))
#else
    #set($myTestParam = "not passed")
#end

Current param value : $myTestParam
```
You might drop this code in the template code of a page.

**Comment by nollymar** [comment]
Passed internal qa

**Comment by bryanboza** [comment]
Fixed, tested based in the provided comment and works as expected

**Comment by jcastro-dotcms** [comment]
The official code changes have been included in the dotCMS 5.3.8.4 LTS release only. The 5.2.8.4 LTS release is several commits behind, which made the code back-port not possible.
